docs/ref/models/querysets.txt

```diff
@@ -264,6 +264,78 @@ control the name of the annotation::
 For an in-depth discussion of aggregation, see :doc:`the topic guide on
 Aggregation </topics/db/aggregation>`.
 
+
+.. currentmodule:: django.db.models.query
+
+.. class:: FilteredRelation(relation_name, condition=Q())
+
+    .. versionadded:: 2.0
+    .. attribute:: FilteredRelation.relation_name
+
+        The name of the field on which you'd like to filter the relation.
+    .. attribute:: FilteredRelation.condition
+
+        A :class:`~django.db.models.Q` object to control the filtering.
+
+Annotate supports also a new kind of argument that is an instance of
+:class:`.FilteredRelation`.
+
+This kind of parameter allows to pass filtering options inside the the ``ON``
+clause when a ``JOIN`` is performed.
+It does not act on the default relationship but on the given ``alias`` given
+as name of the parameter (``pizzas_vegetarian`` in example below).
+
+Example::
+
+    >>> from django.db.models import FilteredRelation
+    >>> Restaurant.objects.annotate(
+    ...    pizzas_vegetarian=FilteredRelation(
+    ...        'pizzas', condition=Q(pizzas__vegetarian=True),
+    ...    )).filter(pizzas_vegetarian__name__icontains='Mozzarella')
+
+This query will join the model Restaurant only to Pizzas that have vegetarian
+flag to True. This results into a performance improvement, noticeable if the
+join is performed through a large number of rows. Because the filtering from
+the ``WHERE`` clause will only operate on a subset of Pizzas (the vegetarian
+ones).
+
+.. admonition:: You need to be aware of some limitations
```

Comment:
I would just title this "Limitations". Pretty much by definition the docs are all about things that the reader has to be aware of, plus the wording seems a bit colloquial

Reply:
agreed